go-controller/pkg/cni/cni.go

```diff
@@ -146,22 +172,38 @@ func (pr *PodRequest) cmdAddWithGetCNIResultFunc(
 			if err = pr.addDPUConnectionDetailsAnnot(kubecli, clientset.podLister, netdevName); err != nil {
 				return nil, err
 			}
-			annotCondFn = isDPUReady
+			// defer DPU connection details check so primary UDN annotation check can be done in parallel if any
```

Comment:
I don't get this comment. Nothing is happening in parallel with the code execution of annotCondFns right? It is just chaining annotationFns together serially to be called? I think you mean here you want to add the primary UDN connection details to the annotation first, before you check DPU is ready for both default and UDN interfaces?

Reply:
assumes default network's isDPUReady() is called here, it will needs to wait for the dpu node consume the default network's connection details annotation and annotate its connection status back. If we move it to the end, other xxxReady() function can continue in parallel